Split sekiban core into Sekiban.Core.Neutral and Sekiban.Core.AspNetCore

info-miura opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the Sekiban Contributing Guide: [https://developers.forem.com/contributing-guide/forem#create-a-pull-request](https://github.com/J-Tech-Japan/Sekiban/blob/main/CONTRIBUTING.md)
     - 📖 Read the Sekiban Code of Conduct: [https://github.com/forem/forem/blob/main/CODE_OF_CONDUCT.md](https://github.com/J-Tech-Japan/Sekiban/blob/main/CODE_OF_CONDUCT.md)
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.

     NOTE: Pull Requests from forked repositories will need to be reviewed by
     a Sekiban Team member before any CI builds will run. Once your PR is approved
     with a `/ci` reply to the PR, it will be allowed to run subsequent builds without
     manual approval.
-->

## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR splits `Sekiban.Core` into `Sekiban.Core.Neutral` and `Sekiban.Core.AspNetCore`.
`Sekiban.Core.Neutral` no longer depends on `Microsoft.AspNetCore.App`, so it can be used in more restricted configurations.

## Related Tickets & Documents

<!--
For pull requests that relate or close an issue, please include them
below.  We like to follow [Github's guidance on linking issues to pull requests](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

For example having the text: "closes #1234" would connect the current pull
request to issue 1234.  And when we merge the pull request, Github will
automatically close the issue.
-->

- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [x] No, and this is why: This PR would only split package and move implementations. Therefore, it might be sufficient to pass existing tests.
- [ ] I need help with writing tests

## [optional] Are there any post deployment tasks we need to perform?
Which implementation of `IUserInformationFactory` should be used without depending on `Microsoft.AspNetCore.App`?

## [optional] What gif best describes this PR or how it makes you feel?

tomohisa reviewed (changes requested):
@info-miura , Thank you for the great work!

After the consideration with @tkawae , we recommend following project structure.

Rename Sekiban.Core.Neutral -> Sekiban.Core.DotNet - and it will be released as nuget
Rename Sekiban.Core.AspNetCore -> Sekiban.Core - and it will be released as nuget

Remove Sekiban.Core - it is same with Sekiban.Core.AspNetCore now.

In future release, current Sekiban.Pure will be main sekiban, and we will consider naming by then.

info-miura commented:
Moved `Sekiban.Core.Neutral` to `Sekiban.Core.DotNet` and `Sekiban.Core.AspNetCore` to `Sekiban.Core`.
And also moved some AspNetCore dependent implementations to `Sekiban.Core`.

tomohisa commented on src/Sekiban.Core/Dependency/SekibanAspNetCoreEventSourcingDependency.cs:
```diff
@@ -159,7 +160,8 @@ SekibanCoreServiceExtensions.MultiProjectionType multiProjectionType
                 Assembly.GetExecutingAssembly(),
-                GetAssembly()));
+                GetAssembly(),
+                typeof(UpdateNoticeEventSubscriber<>).Assembly)); // Sekiban.Core.DotNet needs to be added
```
Sekiban.Core.DotNet のAssemblyが登録されていないため、エラーで落ちていました。追加して、問題がなくなったら許可してマージします。
